spyder/utils/qthelpers.py

```diff
@@ -502,7 +502,42 @@ def show_std_icons():
     dialog.show()
     sys.exit(app.exec_())
 
-    
+
+def create_control_layout(controls, main_widget):

```

Comment:
Well, we agree to disagree, then. :)

In a statically typed language, you would overload the function. Python does not do this because of dynamic typing. IMO `numpy.sin(2)`, `numpy.sin(numpy.array([2, 3]))`, `numpy.sin([2, 3])` would be a comparable case of allowing different types. Anyway, if you don't want this, we'll try something else.

The version with different keyword args (some of which must or must not be used) is worse than the proposed extra logic within the function. You would need the logic to check that a consistent set of arguments is provided. And then the extra logic to decide what to do.

To get things done, I propose to go one step back. I'll leave the creation of the tools layout to the individual dock widgets (as was till now) and only provide `create_control_layout(tools_layout, main_widget)`.